src/core/effects/qgsblureffect.cpp

```diff
@@ -107,11 +112,18 @@ void QgsBlurEffect::readProperties( const QgsStringMap &props )
 
   mEnabled = props.value( QStringLiteral( "enabled" ), QStringLiteral( "1" ) ).toInt();
   mDrawMode = static_cast< QgsPaintEffect::DrawMode >( props.value( QStringLiteral( "draw_mode" ), QStringLiteral( "2" ) ).toInt() );
-  int level = props.value( QStringLiteral( "blur_level" ) ).toInt( &ok );
+  double level = props.value( QStringLiteral( "blur_level" ) ).toDouble( &ok );
   if ( ok )
   {
     mBlurLevel = level;
   }
+  if ( !props.contains( QStringLiteral( "blur_unit" ) ) && props.contains( QStringLiteral( "blur_level" ) ) )
+  {
+    // deal with pre blur unit era by assuming 96 dpi and converting pixel values as millimeters
+    mBlurLevel *= 0.2645;
+  }
+  mBlurUnit = QgsUnitTypes::decodeRenderUnit( props.value( QStringLiteral( "blur_unit" ) ) );
```

Comment:
Does this need a default value (mm)?

Reply:
@nyalldawson , doesn't QgsUnitTypes::decodeRenderUnit() return millimeters by default? That's what the code says unless I'm mistaken.